docs/DeveloperGuide.adoc

```diff
@@ -475,23 +475,34 @@ is rather high. Implementing alternative 2 would be more tedious on the user's s
 
-This List mechanism is facilitated by commands which uses `Logic` interface to obtain the copy of the list that is required. As seen by the diagram 3.2.1 and 3.2.2 above, to obtain the current blacklist of the address book, "list" string is first captured by the `CommandBox`. The `CommandBox` then passes this string to the `Logic` interface for execution.
+The following sequence diagram, Figure 3.5.1, shows further details of the interaction between the user and various application components as a whole, when the blacklist command is executed:
+
+image::BlacklistCommandSequenceDiagram1.PNG[width="800"]
+Figure 3.5.1 : Sequence Diagram of How the Blacklist Command Works_
+
+The following sequence diagram, Figure 3.5.2, shows further details of the interaction between `Model` and 'Addressbook` to obtain the blacklist:
+
+image::BlacklistCommandSequenceDiagram2.png[width="800"]
+Figure 3.5.2 : Sequence Diagram of Model and Addressbook Interacts To Generate The Blacklist From All Persons_
+
+This List mechanism is facilitated by commands which uses `Logic` interface to obtain the copy of the list that is required. As seen by the diagram 3.5.1 and 3.5.2 above, to obtain the current blacklist of the address book, "blacklist" `string` is first captured by the `CommandBox` class. The `CommandBox` class then passes this string to the `Logic` interface for execution.
 
-`Logic` interface uses `LogicManager`, the class that implements its methods, to package it as a `Command` object. The respective command is then executed by this following statement in `LogicManager` class:
+`Logic` interface uses `LogicManager` class to firstly validate the written command and package it as a `Command` object. The respective command is then executed by this following statement in `LogicManager` class:
 
 [source,java]
 ----
 CommandResult result = command.execute();
 ----
 
-Specifically for `blacklist` command, it calls the `getFiltererdBlacklistedPersons()` method residing in the `Logic` interface. `Logic` `Manager` subsequently calls `Model` interface. `ModelManager` handles the command and thereafter calls `ReadOnlyAddressBook` interface to handle the request. `AddressBook` class handles this request and thus will return the required list, since the various lists are stored in this class. Technically, the `List` `Object` that exists in the current structure just contains all persons. The blacklisted persons are obtained by running a check on all persons residing in this variable. This is shown by the following code below:
+Specifically for `blacklist` command, it calls the `getFiltererdBlacklistedPersons()` method residing in the `Logic` interface. `Logic` `Manager` subsequently calls `Model` interface. `Model` uses `ModelManager` class to handle the command and thereafter calls `ReadOnlyAddressBook` interface to handle the request. 'ReadOnlyPersonBook` uses `AddressBook` class to handle this request.
```

Comment:
shouldn't u standardize `LogicManager`? U used `ModelManager` instead of `Model` `Manager`

Reply:
Noted.